### PR TITLE
Add Web Config to public api

### PR DESCRIPTION
**Pull request by antrix1989**

### CODE REVIEW

File: MSAL/src/MSALInteractiveTokenParameters.m [hunk 1]
```diff
@@ -28,21 +28,73 @@
...
 
+- (instancetype)initWithScopes:(NSArray<NSString *> *)scopes
+{
+    return [self initWithScopes:scopes
+                  webviewParameters:[MSALWebviewParameters new]];
+}
+
+#if TARGET_OS_IPHONE
+
+- (void)setParentViewController:(UIViewController *)parentViewController
```
Comment (oldalton) [inline MSAL/src/MSALInteractiveTokenParameters.m]: are we sure this won't be needed for macOS?

File: MSAL/src/public/MSALPublicClientApplication.h [hunk 2]
```diff
@@ -43,45 +43,45 @@
 @interface MSALPublicClientApplication : NSObject
 
 /*!
+    A copy of the configuration which was provided in the initializer.
```
Comment (oldalton) [inline MSAL/src/public/MSALPublicClientApplication.h]: nit: add a warning here that modifying configuration after creation wouldn't change anything?

File: MSAL/src/public/MSALPublicClientApplication.h [hunk 3]
```diff
@@ -103,7 +103,7 @@
     @param  error       The error that occurred creating the application object, if any (optional)
  */
 - (nullable instancetype)initWithClientId:(nonnull NSString *)clientId
```
Comment (oldalton) [inline MSAL/src/public/MSALPublicClientApplication.h]: what's the reason to deprecate this one? It seems simple enough and most of developers are using this one right now. I think Android was planning to keep this one around as well.

File: MSAL/src/public/MSALWebviewParameters.h [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+/*!
+ For a webviewType MSALWebviewTypeWKWebView, custom WKWebView can be passed on.
+ Web content will be rendered onto this view.
+ Observe strings declared in MSALPublicClientStatusNotifications to know when to dismiss.
+ */
+@property (nonatomic, nullable) WKWebView *customWebview;
+
+#if TARGET_OS_IPHONE
+- (nonnull instancetype)initWithParentViewController:(UIViewController *)parentViewController;
```
Comment (oldalton) [inline MSAL/src/public/MSALWebviewParameters.h]: it feels weird for macOS - you need to create new instance and pass it... Seems unnecessary API for macOS if we introduce it. Will macOS also require parent controller? If not, do we want to make this optional for macOS?

File: MSAL/src/public/MSALPublicClientApplication.h (hunk 2 above)
Comment (antrix1989) [reply MSAL/src/public/MSALPublicClientApplication.h]: done

File: MSAL/src/public/MSALPublicClientApplication.h (hunk 3 above)
Comment (antrix1989) [reply MSAL/src/public/MSALPublicClientApplication.h]: done

File: MSAL/src/MSALInteractiveTokenParameters.m (hunk 1 above)
Comment (antrix1989) [reply MSAL/src/MSALInteractiveTokenParameters.m]: Not sure, but [keyWindow](https://developer.apple.com/documentation/appkit/nsapplication/1428406-keywindow) isn't deprecated for MacOS :) We will need to double test this when we start working on MacOS changes.

File: MSAL/src/public/MSALWebviewParameters.h (hunk 4 above)
Comment (antrix1989) [reply MSAL/src/public/MSALWebviewParameters.h]: done

File: MSAL/src/configuration/MSALWebviewParameters.m [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+
+- (id)copyWithZone:(__unused NSZone *)zone
+{
+    MSALWebviewParameters *item;
+#if TARGET_OS_IPHONE
+    item = [[MSALWebviewParameters alloc] initWithParentViewController:_parentViewController];
+    item.parentViewController = _parentViewController;
+    item.presentationStyle = _presentationStyle;
+#else
+    item = [MSALWebviewParameters new];
```
Comment (oldalton) [inline MSAL/src/configuration/MSALWebviewParameters.m]: should we copy parameters for macOS as well?

File: MSAL/src/public/MSALInteractiveTokenParameters.h [hunk 6]
```diff
@@ -62,33 +63,63 @@ NS_ASSUME_NONNULL_BEGIN
...
+
 /*!
  For a webviewType MSALWebviewTypeWKWebView, custom WKWebView can be passed on.
  Web content will be rendered onto this view.
  Observe strings declared in MSALPublicClientStatusNotifications to know when to dismiss.
  */
-@property (nonatomic, nullable) WKWebView *customWebview;
+@property (nonatomic, nullable) WKWebView *customWebview DEPRECATED_MSG_ATTRIBUTE("Create MSALWebviewParameters and provide it to -initWithScopes:webviewParameters: instead");
+
+#if TARGET_OS_IPHONE
```
Comment (oldalton) [inline MSAL/src/public/MSALInteractiveTokenParameters.h]: should this be opposite? if not TARGET_OS_IPHONE?

File: MSAL/src/public/MSALPublicClientApplication.h [hunk 7]
```diff
@@ -54,34 +56,33 @@
...
-@property (readonly, nonnull) MSALAuthority *authority;
+@property (readonly, nonnull) MSALAuthority *authority DEPRECATED_MSG_ATTRIBUTE("Use authority in MSALPublicClientApplicationConfig instead (create your config and pass it to -initWithConfiguration:error:)");
 
 /*! The client ID of the application, this should come from the app developer portal. */
-@property (readonly, nonnull) NSString *clientId;
+@property (readonly, nonnull) NSString *clientId DEPRECATED_MSG_ATTRIBUTE("Use clientId in MSALPublicClientApplicationConfig instead (create your config and pass it to -initWithConfiguration:error:)");
 
 /*! The redirect URI of the application */
-@property (readonly, nonnull) MSALRedirectUri *redirectUri;
+@property (readonly, nonnull) MSALRedirectUri *redirectUri DEPRECATED_MSG_ATTRIBUTE("Use redirectUri in MSALPublicClientApplicationConfig instead (create your config and pass it to -initWithConfiguration:error:)");
```
Comment (oldalton) [inline MSAL/src/public/MSALPublicClientApplication.h]: those are readonly, so maybe remove the note about "(create your config and pass it to -initWithConfiguration:error:)" for readonly ones?

File: MSAL/src/public/MSALInteractiveTokenParameters.h (hunk 6 above)
Comment (antrix1989) [reply MSAL/src/public/MSALInteractiveTokenParameters.h]: Nope, it is correct: we deprecate api for iphone and in "else" we declare similar but not deprecated api for mac.

File: MSAL/src/configuration/MSALWebviewParameters.m (hunk 5 above)
Comment (antrix1989) [reply MSAL/src/configuration/MSALWebviewParameters.m]: These 2 are declared for ios only. Currently, we don't use them on MacOS.

File: MSAL/src/public/MSALPublicClientApplication.h (hunk 7 above)
Comment (antrix1989) [reply MSAL/src/public/MSALPublicClientApplication.h]: done

File: MSAL/src/public/MSALInteractiveTokenParameters.h (hunk 6 above)
Comment (oldalton) [reply MSAL/src/public/MSALInteractiveTokenParameters.h]: ok :)

**Pull request merged**